packages/adapters/src/adapters/Cornerstone/Segmentation_4X.js

```diff
@@ -1541,3 +1660,109 @@ function getSegmentMetadata(multiframe, seriesInstanceUid) {
         data
     };
 }
+
+function readFromUnpackedChunks(chunks, offset, length) {
```

Comment:
Consider using the code from StaticDICOMweb that has all this functionality, exposing it as though it were a single Uint8Array object - that is asyncIterableToBuffer.  That allows reading very large buffers and automatically splitting them as required.  I think we will need that type of functionality in the future for better handling of streaming results so it would be nice to include.  I could add it into dcmjs if that is where we think it belongs.